src/components/Menu.js

```diff
@@ -1,22 +1,42 @@
 /* eslint-disable */
-
-import React from 'react';
-
-const MenuFactory = React.createClass({
-  propTypes: {
-    id: React.PropTypes.string,
-    isOpen: React.PropTypes.bool,
-  },
-
-  toggleMenu() {
-    console.log('toggleMenu ', this.state);
-
-    const newState = { isOpen: !this.state.isOpen };
-
-    console.log(newState);
-
-    this.setState(newState);
-  },
+import React, { Component } from 'react';
+
+import { scrollable } from '../behaviors';
+import { MenuItem } from './Elements';
+
+function MenuInternal(props) {
+  return (
+    <div>
+      <div className="cross-icon">
+        <button onClick={props.toggleMenu} className="ui large button">
+          <i className="download icon"></i>
+          Cross
+        </button>
+      </div>
+      <header>
+        <h1 className="bm-menu-title">Stages</h1>
+      </header>
+      {props.searchField}
+      <nav className="bm-item-list">
+        {props.items}
+      </nav>
+    </div>
+  );
+}
+const ScrollableMenu = scrollable(MenuInternal);
+
+/**
+  * Renders a menu, updating styles on DOM elements outside of this.
+  */
+class MenuFactory extends Component {
+  constructor(props) {
+    super(props);
+    const initialIsOpenProp = props && typeof props.isOpen !== 'undefined';
+
+    this.state = {
+      isOpen: initialIsOpenProp ? this.props.isOpen : false
+    };
+  }
 
   // Sets or unsets styles on DOM elements outside the menu component.
```

Comment:
Yes, this was previous functionality. What are you wanting to refactor? The component's use of state to track whether it is open? Or that an optional prop feeds the state's initial value?